harness/reporters/json.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2017 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package reporters
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+)
+
+type JSONReporter struct {
+	tests  []JSONTest
+	result string
+
+	// values in context will be present in the root level of the
+	// resulting json output
+	context map[string]string
+
+	// non-output related variables
+	filename string
+}
+
+func NewJSONReporter(context map[string]string, filename string) *JSONReporter {
+	return &JSONReporter{
+		context:  context,
+		filename: filename,
+	}
+}
+
+func (r *JSONReporter) MarshalJSON() ([]byte, error) {
+	a := map[string]interface{}{}
```

Comment:
I would rather have a proper struct defined (and possibly exported) for this json structure.